app/controllers/submissions_controller.rb

```diff
@@ -169,19 +147,60 @@ def update_plagiarism
         title: "",
         info: nil)
       sub_comment.save
-      # Copy any grades
-      @submission.grades.each do |g|
-        new_g = Grade.new(
-          grader_id: g.grader_id,
-          submission_id: sub.id,
-          grading_output: g.grading_output,
-          score: g.score,
-          out_of: g.out_of,
-          available: g.available)
-        new_g.save
-      end
     end
-    redirect_to course_assignment_path(@course, @assignment)
+    redirect_to course_assignment_path(@course, @assignment),
+                notice: "Submission marked as plagiarized for #{guilty_students.map{|u| User.find(u).name}.join(', ')}"
+  end
+
+  def split_submission
+    if @submission.users.count == 1
+      redirect_back course_assignment_submission_path(@course, @assignment, @submission),
+                    alert: "Submission is not a team-submission; no need to split it"
+      return
+    end
+    @submission.users.each do |student|
+      split_sub(@submission, student)
+    end
+    redirect_to course_assignment_path(@course, @assignment),
+                notice: "Group submission split for #{@submission.users.map(&:name).join(', ')}"
+  end
+
+  def split_sub(orig_sub, for_user, score = nil)
+    team = orig_sub.team
+    if team
+      # Construct a one-use team, so that this student can be graded in isolation
+      team = Team.new(
+        course: @course,
+        start_date: orig_sub.created_at,
+        end_date: orig_sub.created_at)
+      team.users = [for_user]
+      team.save
+    end
+    # Create the new submission, reusing the prior submitted file
+    sub = Submission.new(
```

Comment:
I thought about that, but since most of the fields are different, and since .dup doesn't preserve created_at (which is one of the fields I want to preserve), it seemed clearer to me to be explicit about the full property set for these objects.